react/features/base/participants/functions.js

```diff
@@ -46,6 +46,59 @@ const AVATAR_CHECKER_FUNCTIONS = [
 ];
 /* eslint-enable arrow-body-style, no-unused-vars */
 
+/**
+ * Returns the list of active speakers that should be moved to the top of the sorted list of participants so that the
+ * dominant speaker is visible always on the vertical filmstrip in stage layout.
+ *
+ * @param {Function | Object} stateful - The (whole) redux state, or redux's {@code getState} function to be used to
+ * retrieve the state.
+ * @returns {Array<string>}
+ */
+export function getActiveSpeakersToBeDisplayed(stateful: Object | Function) {
+    const state = toState(stateful);
+    const {
+        fakeParticipants,
+        sortedRemoteScreenshares,
+        sortedRemoteVirtualScreenshareParticipants,
+        speakersList
+    } = state['features/base/participants'];
+    const { visibleRemoteParticipants } = state['features/filmstrip'];
+
+    // Do not re-sort the active speakers if all of them are currently visible.
+    if (typeof visibleRemoteParticipants === 'undefined' || speakersList.size <= visibleRemoteParticipants.size) {
+        return speakersList;
```

Comment:
This should return a new Map, the calling function expects that.

Reply:
[speakersList](https://github.com/jitsi/jitsi-meet/blob/master/react/features/base/participants/reducer.js#L74) is already a Map and the calling function doesn't modify it. Do we still need to return a new Map in this case ?

Reply:
As discussed, not strictly necessary.